### ISSUE TITLE
Pen thickness seems to be restricted

### ISSUE THREAD

**Issue by ajyoon** [DigiScore/neoscore#14]
It seems that pen thicknesses larger than about 1mm are not respected.

```python
thick_pen = Pen("000000aa", Mm(30))
cosmetically_thin_pen = Pen("#ff0000", ZERO)
Path.straight_line(ORIGIN, None, (Mm(40), ZERO), pen=thick_pen)
Path.straight_line(ORIGIN, None, (Mm(40), ZERO), pen=cosmetically_thin_pen)
```
gives
![pen](https://user-images.githubusercontent.com/15976214/163745527-6d1e13fa-d9d7-497d-87e2-8731c1b07238.png)

As far as I can tell, Qt should support much thicker (if not arbitrarily thick) pens. Maybe there's an issue in the translation layer from neoscore units to Qt thicknesses?

**Comment by Xavman42** [comment]
I started troubleshooting this and may have found the source of the error. It seems as though the clipping bounds are not expanding with the width of the line. When I go into `q_clipping_path.py`, look at the `calculate_bounding_rect` function, and change `bounding_rect.width() - bounding_rect.x() - clip_start_x` to `bounding_rect.width() - bounding_rect.x() - clip_start_x + 1000`, I can draw vertical lines of a slightly greater width.

Here's an example file I've been using to explore this:
```
from helpers import render_example

from neoscore.core import neoscore
from neoscore.core.pen import Pen
from neoscore.core.path import Path
from neoscore.core.units import ZERO, Mm, Unit
from neoscore.core.brush import Brush

neoscore.setup()

#draw an expanding horizontal line
for i in range(100):
   Path.straight_line((Unit(i),Mm(1)), None, (Unit(1+i), ZERO), pen = Pen("000000aa", Unit(i/10)))

#draw an expanding vertical line
for i in range(100):
   Path.straight_line((Mm(1),Unit(i)), None, (ZERO, Unit(1+i)), pen = Pen("000000aa", Unit(i/10)))

#draw some boxes
for i in range(5):
    Path.rect((Mm(41),Mm(20*i)), None, Mm(10), Mm(10), brush = Brush.no_brush(), pen=Pen("000000aa",Mm(1*i)))

render_example("new_ex")
```

**Comment by ajyoon** [comment]
I think you're on to something! We already do try to account for pen thickness in the bounding/clip rect calculations, but looking at this again I found some problems which I just pushed a fix for in 16a10ce. Arbitrarily cranking up the padding does seem to fix the issue, so I think you're right that the rect calculation here is the source of the error. The code here is pretty subtle, involving a lot of translations between coordinate systems in unintuitive ways, so it's not surprising there should be bugs here. I'll keep looking into this.
